client/clientservice/src/state_snapshot_service.cpp

```diff
@@ -202,19 +202,23 @@ static std::shared_ptr<bftEngine::RequestCallBack> getCallbackLambda(const bftEn
   return std::make_shared<bftEngine::RequestCallBack>(callback);
 }
 
-Status StateSnapshotServiceImpl::GetRecentSnapshot(ServerContext* context,
-                                                   const GetRecentSnapshotRequest* proto_request,
-                                                   GetRecentSnapshotResponse* response) {
-  std::chrono::milliseconds timeout = 5s;  // This is default timeout
+void StateSnapshotServiceImpl::setTimeoutFromDeadline(ServerContext* context, std::chrono::milliseconds& timeout) {
   if (context != nullptr) {
-    auto time_left =
-        gpr_time_to_millis(gpr_time_sub(gpr_now(context->raw_deadline().clock_type), context->raw_deadline()));
-    std::chrono::milliseconds curr_timeout{time_left};
-    if (curr_timeout.count() > 0) {
+    auto curr_timeout =
+        std::chrono::duration_cast<std::chrono::milliseconds>(context->deadline() - std::chrono::system_clock::now());
+    if ((curr_timeout.count() > 0) && (curr_timeout.count() <= MAX_TIMEOUT_MS)) {
       timeout = curr_timeout;
+    } else {
+      LOG_WARN(logger_, "Extreme deadline set by client of client service " << KVLOG(curr_timeout.count()));
     }
   }
+}
 
+Status StateSnapshotServiceImpl::GetRecentSnapshot(ServerContext* context,
+                                                   const GetRecentSnapshotRequest* proto_request,
+                                                   GetRecentSnapshotResponse* response) {
+  std::chrono::milliseconds timeout = 30s;  // This is default timeout
```

Comment:
Nit: you can define a static data member for this.

Reply:
Thanks, @yuliasherman , I will avoid static here. But definitely, i will move this inside setTimeoutFromDeadline as you explained in the next comment.